src/Features/Core/Portable/CodeFixes/Configuration/ConfigureSeverityLevelCodeFixProvider.cs

```diff
@@ -0,0 +1,69 @@
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Configuration
+{
+    [ExportSuppressionFixProvider(PredefinedCodeFixProviderNames.ConfigureSeverity, LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    internal sealed partial class ConfigureSeverityLevelCodeFixProvider : ISuppressionOrConfigurationFixProvider
+    {
+        public bool IsFixableDiagnostic(Diagnostic diagnostic)
+            => !diagnostic.IsSuppressed && !SuppressionHelpers.IsNotConfigurableDiagnostic(diagnostic);
+
+        public FixAllProvider GetFixAllProvider()
+            => null;
+
+        public Task<ImmutableArray<CodeFix>> GetFixesAsync(Document document, TextSpan span, IEnumerable<Diagnostic> diagnostics, CancellationToken cancellationToken)
+            => Task.FromResult(GetConfigurations(document.Project, diagnostics, cancellationToken));
+
+        public Task<ImmutableArray<CodeFix>> GetFixesAsync(Project project, IEnumerable<Diagnostic> diagnostics, CancellationToken cancellationToken)
+            => Task.FromResult(GetConfigurations(project, diagnostics, cancellationToken));
+
+        private static ImmutableArray<CodeFix> GetConfigurations(Project project, IEnumerable<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var nestedActions = ArrayBuilder<CodeAction>.GetInstance();
+            var result = ArrayBuilder<CodeFix>.GetInstance();
+            foreach (var diagnostic in diagnostics)
+            {
+                nestedActions.Add(
+                    new SolutionChangeAction(
+                        "None",
+                        solution => ConfigurationUpdater.ConfigureEditorConfig(EditorConfigSeverityStrings.None, diagnostic, project, cancellationToken)));
+                nestedActions.Add(
+                    new SolutionChangeAction(
+                        "Silent",
+                        solution => ConfigurationUpdater.ConfigureEditorConfig(EditorConfigSeverityStrings.Silent, diagnostic, project, cancellationToken)));
+                nestedActions.Add(
+                    new SolutionChangeAction(
+                        "Suggestion",
+                        solution => ConfigurationUpdater.ConfigureEditorConfig(EditorConfigSeverityStrings.Suggestion, diagnostic, project, cancellationToken)));
+                nestedActions.Add(
+                    new SolutionChangeAction(
+                        "Warning",
+                        solution => ConfigurationUpdater.ConfigureEditorConfig(EditorConfigSeverityStrings.Warning, diagnostic, project, cancellationToken)));
+                nestedActions.Add(
+                    new SolutionChangeAction(
+                        "Error",
+                        solution => ConfigurationUpdater.ConfigureEditorConfig(EditorConfigSeverityStrings.Error, diagnostic, project, cancellationToken)));
+
+                var codeAction = new TopLevelConfigureSeverityCodeAction(diagnostic, nestedActions.ToImmutable());
```

Comment:
1. arraybuilders are cheap.  much cleaner to just put hte builder inside the loop, add to it, and do ToImmutableAndFree here.
2. are these items only created when the list is expanded?  or are they created for all fixes across the solution.  If the former, we don't really need to hyperoptimize like this.